R/data.R

```diff
@@ -105,6 +105,73 @@ process_data <- function(data) {
   path
 }
 
+#' Process fitted params for the generate quantities method
+#'
+#' @noRd
+#' @param data If not `NULL`, then either a path to a CSV file compatible with
+#'   CmdStan or a fit object.
+#' @return Path to data file.
+process_fitted_params <- function(fitted_params) {
+  if (is.character(fitted_params)) {
+    paths <- absolute_path(fitted_params)
+  } else if (checkmate::test_r6(fitted_params, classes = ("CmdStanMCMC"))) {
+    if (all(file.exists(fitted_params$output_files()))) {
+      paths <- absolute_path(fitted_params$output_files())
+    } else {
+      draws <- tryCatch( posterior::as_draws_array(fitted_params$draws()),
```

Comment:
This try-catch was added so the error message is better. Otherwise, it says the CSV files don't exist (draws() looks for CSV files if samples not ready).

Reply:
This is for the case when the output_files() do not exist anymore (most likely a fit object was saved and read in the next session). In that case if draws() do not exist we can't do anything else here.

Reply:
Oh I see, that makes sense.